libraries/freertos_plus/aws/greengrass/test/aws_test_greengrass_discovery.c

```diff
@@ -318,12 +320,18 @@ TEST( Full_GGD, GetIPandCertificateFromJSON )
         if( xStatus == pdPASS )
         {
             ulByteRead = 0;
-            xStatus = GGD_JSONRequestGetFile( &xSocket,
-                                              cBuffer,
-                                              ulBufferSize,
-                                              &ulByteRead,
-                                              &xJSONFileRetrieveCompleted,
-                                              ulJSONFileSize );
+            /* Limit loop count to avoid any chance of infinite loops. */
+            ulRequestLoopCounter = 0;
+
+            do
+            {
+                xStatus = GGD_JSONRequestGetFile( &xSocket,
+                                                  &cBuffer[ ulByteRead ],
+                                                  ulBufferSize - ulByteRead,
+                                                  &ulByteRead,
+                                                  &xJSONFileRetrieveCompleted,
+                                                  ulJSONFileSize );
+            } while( ( xStatus == pdPASS ) && ( xJSONFileRetrieveCompleted != pdTRUE ) && ( ulBufferSize - ulByteRead ) > 0 && ( ++ulRequestLoopCounter < ggdTestMAX_REQUEST_LOOP_COUNT ) );
```

Comment:
Is there any way in which `ulBufferSize < ulByteRead`?  Imagine something isn't working properly – should we check to make sure this isn't the case before continuing in this loop?  See [this PR](https://github.com/aws/amazon-freertos/pull/1615) for an example of where not checking for this caused a problem.

Reply:
Good catch, I think checking `ulBufferSize > ulByteRead` is more appropriate since unsigned will always be > 0